pyproject.toml

```diff
@@ -72,7 +72,8 @@ isort = "^5.13.2"
 openapi-spec-validator = "0.2.8"
 
 mkdocstrings = {extras = ["python"], version = ">=0.25.2,<0.28.0"}
-[tool.poetry.dev-dependencies]
+
+[tool.poetry.group.dev.dependencies]
```

Comment:
huh, does this not then you need to actually specify dev to install dev?

Reply:
those two are equivalent since 1.2, it's just the more modern syntax. need to add `poetry install --with dev` to get the dev depdencies. https://python-poetry.org/blog/announcing-poetry-1.2.0/

using the old lock file, it still installs them because they're in the lock file.

removing the lock file and doing "poetry install" still installs them because it creates a new lock file which includes all sub-groups by default even though that's not what you'd expect.

this is why lockfiles suck. the lockfile you commit impacts what other people get installed. so someone would have to opt out by doing like `poetry install --without dev` to note get the dev packages installed.

Reply:
added `--with dev` in 952cfe010f71b3ce16aebaffc897e80139a5aaaa even though it makes no practical difference as long as the lock file is generated with dev

Reply:
i think its itsd --root right? 
Then you dont get the dev dependencies?

Reply:
reverted that and the original change in e7fcb256743f2c15f9beb5bf0cc3c66f4e718da5 since that wasn't the point of this PR. the thought was to get rid of that deprecation warning, but that can be done separately.